### ISSUE TITLE
How do I save inspect element changes permanently in Chrome?

### ISSUE THREAD

**Issue by Brown946** [Ruud14/Page-Manipulator#15]
Hi please, I'd like to change PERMANENTLY the value 1.47 but I don't know the css code to put on "page manipulator" chrome extension, can help someone please ?

I changed value but only in my screen but not permanently, refresh page shows value 1.47 not my own value I changed before which was 983 899,54

`<div class="tv-account-manager__data js-collateral-value">1.47</div>`

**Comment by Ruud14** [comment]
You don't need any CSS for this, use JavaScript instead.

In Page Manipulator create a JavaScript file,
1. Check the active checkbox.
2. Put the URL of the website in the 'Active websites' textarea.
3. Set 'Matching pages' to recursive.
4. Put this code in the file:
```javascript
// Find all elements with 'tv-account-manager__data' as one of its class names.
var elementsWithThisClassName = document.getElementsByClassName("tv-account-manager__data");
// Check all those elements to see whether their value is "983 899,54".
for (var i = 0; i < elementsWithThisClassName.length; i++) {
    // If that is the case we change it to "1.47"
    if (elementsWithThisClassName[i].innerHTML  == "983 899,54") {
        elementsWithThisClassName[i].innerHTML = "1.47"
    }
}
```
I hope the comments (indicated by the grey lines starting with '//') can help you understand the code.

**Comment by Brown946** [comment]
hi Ruud14:  thank you so much for explications - I understand good the code

I did exactly what you wrote but same result after "RELOAD" on "page manipulator" - the website showed 1.47 value..

Could I send you by mail .pdf file for good understand what I meaned ?

**Comment by Ruud14** [comment]
Sure, go ahead. My email is on my main Github page.

**Comment by Miekviper1** [comment]
hi! I have an incremental game where there are values that I cannot find and/or change... Could someone help me with understanding how to change these values?

**Comment by Ruud14** [comment]
Whether or not it is possible to change the values depends on how the game is built. Please elaborate on the game and the values you're trying to change, and I might have some time to look at it.

**Comment by Miekviper1** [comment]
Hiya, Here are  the two games, quite similar in theme, I'm not quite sure
if it differs in code. fishgame
<https://coshxx.github.io/fishgame/#/tab/fish/fish>, fg2share
<https://coshxx.github.io/fg2share/>. What I would like to change is the
value/number of the tickspeed.
This could be the rate or time of the fishing session on the second game or
the tickrate of the office worker and fish per second in the first game.

I've been searching around the console and exploring google to see if
anyone had 'decoded' the save, but this has never come to a
satisfactory conclusion.

Would you be able to help? It would be great if you could and if you could
shed some light on changing code could also help me personally.

Thanks a lot!
Mieky.

**Comment by Miekviper1** [comment]
Hi, have you been able to look at these two games? thanks.

**Comment by Ruud14** [comment]
Yes, I have.
I forgot about it as you had commented this on a closed issue, my apologies.

Usually, when trying to change something in such a browser game, one would look at the javascript code of said game to find variables worth changing.
This can be done by opening the browser devtools and then going to 'sources'.   
Once said variables have been found, one could go to the 'console' and try manually changing those variables and see if it really produces the desired output. 

For these two games however, it is more complicated as the javascript has been obfuscated. This means that the author has purposefully made the code difficult to read for people like you and me by removing the overal structure of the code and replacing all variable names with nonsense.

There exist tools like [these](https://deobfuscate.io/) to deobfuscate the code but this doesn't get rid of the nonsense variable names. I'd suggest you to watch [this](https://www.youtube.com/watch?v=Ka8PXERJ-Fs) to learn more about deobfuscation. 

If you eventually manage to find variables to change, you could automate changing those variables by injecting a piece of javascript code using this chrome extension. 

TL;DR The developer has made it quite difficult for you to find variables to change. This doesn't make it impossible, but you'll have to put in a lot of work and trial and error to find the right variables to change. After finding the variables, changing them could be automated using this extension. 

Hope this helps,

Ruud Brouwers

**Comment by Miekviper1** [comment]
All good. Thanks, I'll check it out. Also, did you link the chrome
extension? sorry if you did and i'm just blind lol. If not please let me
know what it is. Thanks!!

**Comment by Ruud14** [comment]
You're commenting this on the issue page of the extension that we're talking about.

**Comment by Miekviper1** [comment]
Sorry, I was replying through email!

**Comment by Tithe108** [comment]
Hello I am trying to change this text on a grade I have and I have tried the chrome extension changer and it is not working. If I could have any sort of help it will help me very much in not getting kicked out. If you find it more comfortable talking through email mine is o.bradley00@gmail.com

**Comment by Ruud14** [comment]
Changes will only be visible on your computer, so unless your grade is being looked at on your screen, it will unfortunately not work.

**Comment by yousaaf** [comment]
How to save the elements in inspect element on youtube video
